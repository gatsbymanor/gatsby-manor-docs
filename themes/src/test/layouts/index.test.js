import React from 'react';
import TemplateWrapper from '../../src/layouts/index';

describe('<TemplateWrapper />', () => {
  const testData = {
    location: {
      pathname: '/home'
    }
  };

  const children = jest.fn();

  const component = shallow(
    <TemplateWrapper
      location={testData.location}
      children={children} />
  );


  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render have the right props', () => {
    const props = component.instance().props;
    expect(props).toHaveProperty('children');
    expect(props).toHaveProperty('location');
  });
});
