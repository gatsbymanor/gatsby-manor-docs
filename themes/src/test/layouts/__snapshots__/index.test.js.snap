// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateWrapper /> should render correctly 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    meta={
      Array [
        Object {
          "content": "https://storage.googleapis.com/gatsby_manor_assets/gatsbymanor_og_image.jpg",
          "property": "og:image",
        },
        Object {
          "content": "https://gatsbymanor.com",
          "property": "og:url",
        },
        Object {
          "content": "Themes for GatsbyJS",
          "property": "og:title",
        },
        Object {
          "content": "Browse our gallery of themes. Use the theme with GatsbyJS.",
          "property": "og:description",
        },
      ]
    }
    title="Gatsby Manor, themes for GatsbyJS"
  />
  <Navbar
    enableSideMenu={[Function]}
    location={
      Object {
        "pathname": "/home",
      }
    }
  />
  <div
    className=""
  />
  <Footer
    path="/home"
  />
</div>
`;
